File: src/sketchview.cpp
#include "sketchview.h"

SketchView::SketchView(QWidget *parent)
        : QGraphicsView(parent) {
}

SketchView::SketchView(QGraphicsScene *scene, QWidget *parent)
        : QGraphicsView(scene, parent) {
}

SketchView::~SketchView() {
}

void SketchView::focusOutEvent(QFocusEvent *event) {
    QGraphicsView::focusOutEvent(event);
    QCursor cursor;
    cursor.setShape(Qt::CustomCursor);
    setCursor(cursor);
    emit hideFromTop();
}

void SketchView::focusInEvent(QFocusEvent *event) {
    QGraphicsView::focusInEvent(event);
    QCursor cursor;
    cursor.setShape(Qt::OpenHandCursor);
    setCursor(cursor);
    emit showAtTop();
}

void SketchView::mousePressEvent(QMouseEvent *event) {
    QGraphicsView::mousePressEvent(event);
    QCursor cursor;
    cursor.setShape(Qt::ClosedHandCursor);
    setCursor(cursor);
}

void SketchView::mouseReleaseEvent(QMouseEvent *event) {
    QGraphicsView::mouseReleaseEvent(event);
    QCursor cursor;
    cursor.setShape(Qt::OpenHandCursor);
    setCursor(cursor);
}

File: python/sketchview.py
from PySide2.QtCore import Qt, Signal
from PySide2.QtGui import QCursor
from PySide2.QtWidgets import QGraphicsView


class SketchView(QGraphicsView):
    def __init__(self):
        super(SketchView, self).__init__()

    def focusOutEvent(self, event):
        super(SketchView, self).focusOutEvent(event)
        cursor = QCursor()
        cursor.setShape(Qt.CustomCursor)
        self.setCursor(cursor)
        self.hideFromTop.emit()

    def focusInEvent(self, event):
        super(SketchView, self).focusInEvent(event)
        cursor = QCursor()
        cursor.setShape(Qt.OpenHandCursor)
        self.setCursor(cursor)
        self.showAtTop.emit()

    def mousePressEvent(self, event):
        super(SketchView, self).mousePressEvent(event)
        cursor = QCursor()
        cursor.setShape(Qt.ClosedHandCursor)
        self.setCursor(cursor)

    def mouseReleaseEvent(self, event):
        super(SketchView, self).mouseReleaseEvent(event)
        cursor = QCursor()
        cursor.setShape(Qt.OpenHandCursor)
        self.setCursor(cursor)

    @Signal
    def hideFromTop(self):
        pass

    @Signal
    def showAtTop(self):
        pass
